python/lgbserver/lgbserver/model.py

```diff
@@ -57,17 +60,13 @@ def load(self) -> bool:
 
     def predict(self, payload: Union[Dict, InferRequest], headers: Dict[str, str] = None) -> Union[Dict, InferResponse]:
         try:
-            if isinstance(payload, Dict):
-                dfs = []
-                instances = payload['inputs'] if "inputs" in payload else payload['instances']
-                for input in instances:
-                    dfs.append(pd.DataFrame(input, columns=self._booster.feature_name()))
-                inputs = pd.concat(dfs, axis=0)
-                result = self._booster.predict(inputs)
-                return {"predictions": result.tolist()}
-            else:
-                instance = get_predict_input(payload)
-                result = self._booster.predict(instance)
-                return get_predict_response(payload, result, self.name)
+            dfs = []
+            instances = get_predict_input(payload)
+            for input in instances:
+                dfs.append(pd.DataFrame(
+                    input, columns=self._booster.feature_name()))
+            inputs = pd.concat(dfs, axis=0)
```

Comment:
we need a check for pandas type here, as if the user is sending in pd as content type then it is already pandas dataframe

Reply:
made a commit to handle this in the `get_predict_inputs`